tests/src/test/spark330/scala/com/nvidia/spark/rapids/BloomFilterAggregateQuerySuite.scala

```diff
@@ -65,74 +68,149 @@ class BloomFilterAggregateQuerySuite extends SparkQueryCompareTestSuite {
         (1L to 100L).map(_ => None)).toDF("col")
   }
 
-  for (numEstimated <- Seq(4096L, 4194304L, Long.MaxValue,
+  private def withExposedSqlFuncs[T](spark: SparkSession)(func: SparkSession => T): T = {
+    try {
+      installSqlFuncs(spark)
+      func(spark)
+    } finally {
+      uninstallSqlFuncs(spark)
+    }
+  }
+
+  private def doBloomFilterTest(numEstimated: Long, numBits: Long): DataFrame => DataFrame = {
+    df =>
+      val table = "bloom_filter_test"
+      val sqlString =
+        s"""
+           |SELECT might_contain(
+           |            (SELECT bloom_filter_agg(col,
+           |              cast($numEstimated as long),
+           |              cast($numBits as long))
+           |             FROM $table),
+           |            col) positive_membership_test,
+           |       might_contain(
+           |            (SELECT bloom_filter_agg(col,
+           |              cast($numEstimated as long),
+           |              cast($numBits as long))
+           |             FROM values (-1L), (100001L), (20000L) as t(col)),
+           |            col) negative_membership_test
+           |FROM $table
+          """.stripMargin
+      df.createOrReplaceTempView(table)
+      withExposedSqlFuncs(df.sparkSession) { spark =>
+        spark.sql(sqlString)
+      }
+  }
+
+  private def getPlanValidator(exec: String): (SparkPlan, SparkPlan) => Unit = {
+    def searchPlan(p: SparkPlan): Boolean = {
+      if (ExecutionPlanCaptureCallback.didFallBack(p, exec)) {
```

Comment:
nit, this could be simplified a bit:
```
ExecutionPlanCaptureCallback.didFallBack(p, exec) || 
  p.children.exists(searchPlan) || 
  p.subqueries.exists(searchPlan)
```